src/inout.c

```diff
@@ -66,6 +67,15 @@ int H2T_loadNpy(MMG5_pMesh mmgMesh, int** tabhex, char* filename) {
     return -1;
   }
 
+  /* Read data type*/
+  fseek(inm,0,0);
+  while (!(buffer == 0x3c)) {
+    fread(&buffer,sizeof(buffer),1,inm);
+  }
+  fread(&buffer,sizeof(buffer),1,inm);
+  fread(&buffer,sizeof(buffer),1,inm);
+  sscanf(&buffer, "%lu", &dataSize);
+
   /* Reach end of header */
   while (!(buffer == 0x0a)) {
```

Comment:
I propose to use a C macro to make understanding easier : `#define H2T_HEX_LINE_FEED`